### PR TITLE
docs: fix SSG explanation

### PR DESCRIPTION
**Pull request by flogehring**
## Description of Problem

The docs state that with SSG data may not change between deploys.

## Proposed Solution

Rephrase it to say data may not change between requests instead.

### CODE REVIEW

**Comment by AloisSeckar** [comment]
I believe the current wording is correct. "Static" means that no matter how many **_requests_** you do, you'll always receive the very same file(s). To change anything in the content, you need to **deploy** a new version of application with updated files.

Purely static sites should be HTML + CSS + media files only. No database, no web services, no processing on server... 

But I think the term is also often used for JavaScript files, that are "static" from the server perspective (they are always served the same), but dynamic for the user - JavaScript on client allows interaction with the content. And this might cause the confusion.

**Comment by flogehring** [comment]
> Purely static sites should be HTML + CSS + media files only. No backend, no database, no web services, no processing on server...

They **are** being processed on the server (or in CI). During **G**eneration in the build process, which may very well include backends and databases etc.  Purely static files are the results of that with SSG. 

The current text said, that the data cannot change between deploys. That sentence would only make sense to me, if you separate the meaning of the build process from the actual deployment as in hosting. If so, the wording should be updated further, to include that notion and thus make it more clear.

Considering the possibly dynamic data at build time I still think that my PR makes more sense than the current wording.

**Comment by AloisSeckar** [comment]
If you make request to `/foo` on SSG site 1000 times, you will receive the very same `foo.html` every time. Data are not changed between **requests**. To change the content, you have to statically build and **deploy** a new version of the application.

**Comment by flogehring** [comment]
Yes absolutely! 😊 

That is why the previous wording was wrong.

**Comment by AloisSeckar** [comment]
Oh, now I am finally getting you. Both sentences are true, but the updated one sounds more explanatory. You're right, sorry.

**Comment by flogehring** [comment]
> Oh, now I am finally getting you. Both sentences are true, but the updated one sounds more explanatory. You're right, sorry.

Glad we're on the same page.

Do you think it would help to change `consuming` into `providing` in the previous sentence?


>The keyword here is **static**: SSG can only be applied to pages ~~consuming~~ providing static data i.e. data that is known at build time and can not change between requests.

**Comment by AloisSeckar** [comment]
> 
> Do you think it would help to change `consuming` into `providing` in the previous sentence?
> 
> > The keyword here is **static**: SSG can only be applied to pages ~consuming~ providing static data i.e. data that is known at build time and can not change between requests.

Now when you point it out, yes. Although we can say a site "consumes" input data to create a content, more natural way of thinking about the concept probably would be to say a site "provides" data to the user...
